x-pack/plugins/apm/server/routes/fleet/get_package_policy_decorators.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { cloneDeep, get, set } from 'lodash';
+import { NewPackagePolicy } from '@kbn/fleet-plugin/common';
+import { AgentConfiguration } from '../../../common/agent_configuration/configuration_types';
+import { AGENT_NAME } from '../../../common/es_fields/apm';
+import { ArtifactSourceMap } from './source_maps';
+
+// agent config
+export const AGENT_CONFIG_PATH = `inputs[0].config['apm-server'].value.agent_config`;
+export const AGENT_CONFIG_API_KEY_PATH = `inputs[0].config['apm-server'].value.agent.config.elasticsearch.api_key`;
+
+// source map
+export const SOURCE_MAP_API_KEY_PATH = `inputs[0].config['apm-server'].value.rum.source_mapping.elasticsearch.api_key`;
+export const SOURCE_MAP_PATH = `inputs[0].config['apm-server'].value.rum.source_mapping.metadata`;
+
```

Comment:
This approach is a lot cleaner than previous. I like using lodash `set` with the path to set deep values.

Reply:
Thanks for noticing. I like it too!